client/dist/index.html

```diff
@@ -7,7 +7,10 @@
     <link rel="stylesheet" href="styles.css"></link>
   </head>
   <body>
-    <main><div id="root"></div></main>
+    <main>
+      <div id="modal"></div>
+      <div id="root"></div>
```

Comment:
As with anything, there's so many correct ways to make a modal, but I believe using createPortal is actually the best way to go about it. createPortal allows for the modal to be rendered outside the flow of the page and outside whatever containers it may be in. This logically makes sense to be used for elements like modals/popups. 

I think it's for the best for clarity that modals get rendered into a separate DOM node.

Reply:
> interesting, create a 'modal' element outside of 'root' element in the index.html. In index.jsx, we rendered the 'App' component within the 'root' element in the HTML file. So i'm curious if 'modal' is outside of 'root' element, can it get rendered?

the cool feature with `createPortal` is that the node that you're rendering outside of the tree, in this case, the react `modal` component, has all the properties of a child and thus, can access the context of its parent nodes as if it were rendered regularly with in `"App"` div.